src/Cell.cpp

```diff
@@ -764,6 +615,16 @@ void Cell::setName(const char* name) {
 
 
 /**
+ * @brief Sets the name of the Cell
+ * @param name the Cell name string
+ */
+void Cell::setRegion(Region* region) {
+  // FIXME: this is a memory leak
+  _region = region;
```

Comment:
I've opted for the latter option. The one downside of this is that the `Region` object given to a `Cell` cannot be change or manipulated after the assignment, since any changes made to it will not be reflected in the copied `Region` stored by the `Cell`. I've documented this though.